torchtext/experimental/datasets/language_modeling.py

```diff
@@ -29,27 +31,22 @@ class LanguageModelingDataset(torch.utils.data.Dataset):
 
     """
 
-    def __init__(self, data, vocab):
+    def __init__(self, data, vocab, transforms):
         """Initiate language modeling dataset.
 
         Arguments:
             data: a tensor of tokens. tokens are ids after
                 numericalizing the string tokens.
                 torch.tensor([token_id_1, token_id_2, token_id_3, token_id1]).long()
             vocab: Vocabulary object used for dataset.
-
-        Examples:
-            >>> from torchtext.vocab import build_vocab_from_iterator
-            >>> data = torch.tensor([token_id_1, token_id_2,
-                                     token_id_3, token_id_1]).long()
-            >>> vocab = build_vocab_from_iterator([['language', 'modeling']])
-            >>> dataset = LanguageModelingDataset(data, vocab)
+            transforms: Text string transforms.
 
         """
 
         super(LanguageModelingDataset, self).__init__()
-        self.data = data
         self.vocab = vocab
+        self.transforms = transforms
+        self.data = torch.cat(tuple(transforms(row) for row in data), axis=0)
```

Comment:
To be consistent with text classification datasets, please call the transform func in `__getitem__` func

Reply:
It might be hard to call the transform func in `__getitem__`. For dataset like WikiText2, the raw data `wiki.train.tokens` is stored as a multi-line txt file, and `self.__getitem__[i]` is expeced to output i-th token. Therefore, pre-processing is required to concat and tokenize the multi-line file. Can you advise if we need an additional `preprocess` func?

Reply:
I cant' move `transforms` into `__getitem__`, because `tokenizer` must be applied before `__getitem__`. Otherwise, the unit test https://github.com/pytorch/text/blob/master/test/data/test_builtin_datasets.py#L54-L57 will fail here.

Possible solution
- Move `tokenizer` out of `transforms` 
- OR: split `transforms` into two parts - global transforms / token-level transforms. 

Please advise how I can move forward. Thanks.